server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -81,8 +81,10 @@ class ArrivalsGraphStage(name: String = "",
       grab(arrivalsInlet) match {
         case ArrivalsFeedSuccess(Flights(flights), connectedAt) =>
           log.info(s"Grabbed ${flights.length} arrivals from connection at ${connectedAt.toISOString()}")
-          handleIncomingArrivals(sourceType, flights)
-          mergeAllSourcesAndPush(baseArrivals, forecastArrivals, liveArrivals)
+          if (flights.nonEmpty || sourceType == BaseArrivals) {
```

Comment:
remove the BaseArrivals condition - double check there's not diffing stage on the base arrivals